src/cogent3/core/new_alignment.py

```diff
@@ -912,6 +1007,378 @@ def dotplot(
             )
         return dotplot
 
+    @UI.display_wrap
+    def apply_pssm(
+        self,
+        pssm: PSSM = None,
+        path: str = None,
+        background: numpy.array = None,
+        pseudocount: int = 0,
+        names: list = None,
+        ui=None,
+    ) -> numpy.array:
+        """scores sequences using the specified pssm
+
+        Parameters
+        ----------
+        pssm :
+            A profile.PSSM instance, if not provided, will be loaded from path
+        path
+            path to either a jaspar or cisbp matrix (path must end have a suffix
+            matching the format).
+        background
+            background frequencies distribution
+        pseudocount
+            adjustment for zero in matrix
+        names
+            returns only scores for these sequences and in the name order
+        ui
+            argument to control the display of a progress bar. i.e.,
+            'show_progress=True'
+
+
+        Returns
+        -------
+        numpy array of log2 based scores at every position
+        """
+        assert not self.is_ragged(), "all sequences must have same length"
+        from cogent3.parse import cisbp, jaspar
+
+        assert pssm or path, "Must specify a PSSM or a path"
+        assert not (pssm and path), "Can only specify one of pssm, path"
+
+        if isinstance(names, str):
+            names = [names]
+
+        if path:
```

Comment:
add a comment to redesign. This method is handling possible cases which would be better delegated to a new function called `load_pssm()` whch should be somewhere else than here.